pkg/admission/webhookconfig.go

```diff
@@ -0,0 +1,66 @@
+package admission
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+// PatchCABundleOnValidatingWebhooks patches the CA bundle of all validating webhook configurations that have the specified labels in the cluster.
+// Webhook configurations should have the following labels:
+// "grafana.com/inject-rollout-operator-ca": "true",
+// "grafana.com/namespace":                  <specified namespace>,
+func PatchCABundleOnValidatingWebhooks(ctx context.Context, logger log.Logger, kubeClient kubernetes.Interface, namespace string, caPEM []byte) error {
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{
+		"grafana.com/inject-rollout-operator-ca": "true",
+		"grafana.com/namespace":                  namespace,
+	}}
+
+	whcs, err := kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(ctx, metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list validating webhook configurations: %w", err)
+	}
+	if len(whcs.Items) == 0 {
+		level.Info(logger).Log("msg", "no validating webhook configurations found")
+		return nil
+	}
+
+	for _, wh := range whcs.Items {
+		level.Info(logger).Log("msg", "found validating webhook configuration", "name", wh.GetName())
```

Comment:
maybe also logging the namespace here wouldn't be bad. You can take a similar approach of having a new logger with these fields. `logger = log.With(logger, "namespace", wh.GetNamespace(), ...)`

Reply:
nit: you can make the same change with the namespace in https://github.com/grafana/rollout-operator/blob/705d27813ee660b01b6d4bb3afe2c147f5963dd7/pkg/admission/no_downscale.go#L26

Reply:
Done in https://github.com/grafana/rollout-operator/pull/25/commits/737e9fac29de012de53c0949fa0b4e056c560b88